climada/util/coordinates.py

```diff
@@ -85,6 +85,65 @@
 are not considered."""
 
 
+def check_if_geo_coords(lat, lon):
+    """Check if latitude and longitude arrays are likely in geographic coordinates,
+    testing if min/max values are within -90 to 90 for latitude and -540 to 540
+    for longitude. Lon coordinates of <-360 or >360 are allowed to cover cases
+    of objects being defined close to the 180 meridian for which longitude
+    intervals such as [179, 538] or [-538, -179] might be used.
+
+    Parameters
+    ----------
+    lat, lon : ndarrays of floats, same shape
+        Latitudes and longitudes of points.
+
+    Returns
+    -------
+    test : bool
+        True if lat/lon ranges seem to be in the geographic coordinates range, otherwise False.
+    """
+    lat = np.array(lat)
+    lon = np.array(lon)
+
+    # Check if latitude is within -90 to 90 and longitude is within -540 to 540
+    # and extent are smaller than 180 and 360 respectively
+    test = (
+        lat.min() >= -91 and lat.max() <= 91 and lon.min() >= -541 and lon.max() <= 541
+    ) and ((lat.max() - lat.min()) <= 181 and (lon.max() - lon.min()) <= 361)
+    return bool(test)
+
+
+def get_crs_unit(coords):
+    """
+    Retrieve the unit of measurement for the coordinate reference system (CRS).
+
+    Parameters
+    ----------
+    coords : GeoDataFrame
+        An object with a coordinate reference system (CRS) attribute.
+
+    Returns
+    -------
+    unit : str
+        The unit of measurement for the coordinate system, as specified in the
+        CRS axis information. Assumes that both axes have the same unit.
+    """
+
+    unit = coords.crs.axis_info[0].unit_name  # assume both axes have the same unit
+    if unit == "degree":
+        pass
+    elif unit == "metre":
+        unit = "m"
+    elif unit == "kilometre":
+        unit = "km"
+    else:
+        raise ValueError(
+            f"Unknown unit: {unit}. Please provide a crs that has a unit "
+            "of 'degree', 'metre' or 'kilometre'."
```

Comment:
Is there a reason to forbid other units? I am not sure to understand why at this stage.

Reply:
How about you raise a warning when the unit is not among the supported ones in the assign centroids part and explain that the threshold might be unsuitable? 

I like the suggestion for improving the default value of the threshold, but maybe this should be done in a separate PR?